src/features/embalm/stepContent/hooks/useCreateSarcohpagus.ts

```diff
@@ -0,0 +1,111 @@
+import { BigNumber, ethers } from 'ethers';
+import { privateKeys } from 'lib/mocks/privateKeys';
+import { encrypt, readFileDataAsBase64 } from 'lib/utils/helpers';
+import { useCallback, useEffect } from 'react';
+import { split } from 'shamirs-secret-sharing-ts';
+import { setIsUploading } from 'store/bundlr/actions';
+import { setPayloadTxId, setSelectedArchaeologists, setShardsTxId } from 'store/embalm/actions';
+import { useDispatch, useSelector } from 'store/index';
+import { Archaeologist } from 'types/index';
+import { useBundlr } from './useBundlr';
+
+async function encryptShards(publicKeys: string[], payload: Uint8Array[]): Promise<Buffer[]> {
+  return Promise.all(publicKeys.map(async (key, i) => encrypt(key, Buffer.from(payload[i]))));
+}
+
+export function useCreateSarcophagus() {
+  const dispatch = useDispatch();
+  const {
+    recipient,
+    file,
+    outerPublicKey,
+    outerPrivateKey,
+    selectedArchaeologists,
+    payloadTxId,
+    shardsTxId,
+  } = useSelector(x => x.embalmState);
+  const { isUploading } = useSelector(x => x.bundlrState);
+  const { uploadFile } = useBundlr();
+
+  // TODO: Move this into its own hook and check all fields
+  const canCreateSarcophagus = recipient.publicKey !== '' && !!outerPublicKey && !!file;
+
+  /**
+   * Double encrypts the payload and uploads it to the bundlr node. The contents of this will go
+   * into the useEffect but is left out until sarcophagus creation is included
+   */
+  const handleSubmit = useCallback(async () => {
+    try {
+      // Prepare the payload
+      if (recipient.publicKey === '' || !file || !outerPublicKey) return;
+      const payload = await readFileDataAsBase64(file);
+
+      // Step 1: Encrypt the inner layer
+      const encryptedInnerLayer = await encrypt(recipient.publicKey, payload);
+
+      // Step 2: Encrypt the outer layer
+      const encryptedOuterLayer = await encrypt(outerPublicKey, encryptedInnerLayer);
+      if (!encryptedOuterLayer) return;
+
+      // Step 3: Split the outer layer private key using shamirs secret sharing
+      const shards: Uint8Array[] = split(outerPrivateKey, {
+        // TODO: Use totalArchaeologists value instead of selectedArchaeologists.length
+        shares: selectedArchaeologists.length,
+        // TODO: Use requiredArchaeologists value
+        threshold: 3,
+      });
+
+      // Step 4: Encrypt each shard of the outer layer private key using each archaeologist's public
+      // key
+      const archPublicKeys = selectedArchaeologists.map(x => x.publicKey || '');
```

Comment:
We will eventually want a check somewhere that ensures each selected archaeologist public key has been set prior to this `handleSubmit` being called. This can get handled later, just wanted to call it out.